uk.co.compucorp.civicrm.tasksassignments/js/test/controllers/task_test.js

```diff
@@ -0,0 +1,56 @@
+define([
+  'common/angularMocks',
+  'tasks-assignments/app'
+], function () {
+  'use strict';
+
+  describe('TaskCtrl', function () {
+    var $scope, $httpBackend;
+
+    beforeEach(module('civitasks.appDashboard'));
+    beforeEach(inject(function ($controller, $rootScope, _$httpBackend_) {
+      $scope = $rootScope.$new();
+      $httpBackend = _$httpBackend_;
+
+      $httpBackend.whenGET(/action=getoptions&entity=Task/).respond({});
+      $httpBackend.whenGET(/action=getoptions&entity=Document/).respond({});
+      $httpBackend.whenGET(/action=get&entity=CaseType/).respond({});
+      $httpBackend.whenGET(/action=get&entity=contact/).respond({});
+
+      setup();
+
+      $controller('TaskCtrl', {
+        $scope: $scope
+      });
+    }));
+
+    describe("Task date", function () {
+      beforeEach(function () {
+        spyOn($scope, 'updateTask');
+        $scope.$digest();
+      });
+
+      it("Update task date", function () {
+        $scope.task.activity_date_time = new Date();
+        $scope.$digest();
+        expect($scope.updateTask).toHaveBeenCalled();
+      });
+    });
+
+    function setup() {
+      $scope.task = {
+        "activity_date_time": "2017-01-10T12:26:00.000Z",
+        "activity_type_id": "70",
+        "id": "1",
+        "status_id": "1",
+        "assignee_contact_id": [],
+        "source_contact_id": "209",
+        "target_contact_id": ["206"],
+        "case_id": ""
+      };
+
+      $scope.updateTask = function () {};
```

Comment:
I think you used it because `updateTask`is undefined. You can always use 
`$scope.updateTask = jasmine.createSpy('updateTask')` in the `beforeEach`